engine/paima-build-utils/src/middleware-esbuildconfig.template.cts

```diff
@@ -39,11 +38,12 @@ const fsaReplace: esbuild.Plugin = {
 
 export const config: esbuild.BuildOptions = {
   // JS output from previous compilation step used here instead of index.ts to have more control over the TS build process
+  // however, that means type definitions will be lost until this is implemented: https://github.com/evanw/esbuild/issues/95#issuecomment-1559710310
   entryPoints: ['build/index.js'],
   bundle: true,
   format: 'esm',
   define,
   outfile: 'packaged/middleware.js',
-  plugins: [polyfillNode({}), dtsPlugin(), fsaReplace],
+  plugins: [polyfillNode({}), fsaReplace],
```

Comment:
This package doesn't work with typescript v5, and upon closer inspection this package actually did nothing anyway because the build script was changed from operation on `ts` files to operating on `js` files. I submitted a fix to chess so that we don't need this anymore anyway, and linked a ticket above that would be a cleaner fix